src/pages/Cinemas/CinemasContainer.jsx

```diff
@@ -1,11 +1,92 @@
+import { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  Container,
+  Grid,
+  Typography,
+  InputLabel,
+  Select,
+  MenuItem,
+  FormHelperText,
+  FormControl,
+} from "@material-ui/core";
 import Cinemas from "./Cinemas";
+import { selectCinemaData } from "../../store/cinema/selectors";
+import { selectCityData } from "../../store/city/selectors";
+import { selectUserData } from "../../store/user/selectors";
+import { cinemaFetch } from "../../store/cinema/asyncActions";
+import { cityFetch } from "../../store/city/asyncActions";
 import Header from "../../components/Header/HeaderContainer";
 
 const CinemasContainer = () => {
+  const userData = useSelector(selectUserData);
+  const cinemasArray = useSelector(selectCinemaData);
+  const citiesArray = useSelector(selectCityData);
+  const [currentCityId, setCurrentCityId] = useState("");
+  const [filteredCinemas, setFilteredCinemas] = useState(cinemasArray);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(cinemaFetch(cinemasArray));
+    dispatch(cityFetch(citiesArray));
+  }, []);
+  useEffect(() => {
+    if (!currentCityId) {
+      return setFilteredCinemas(cinemasArray);
```

Comment:
`useEffect` should return a callback which **unsubscribe current component from any subscriptions** only
it's better to use else here;